src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,73 @@
-     */
+    private static final int ZERO_NUMBER = 0;
+    private static final int SUBTRACT_NUMBER = 1;
+    private static final int CHECKING_NUMBER = 2;
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return Integer.parseInt(numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
+                .filter(n -> Integer.parseInt(n) % CHECKING_NUMBER == 0)
+                .min(Comparator.comparingInt(Integer::parseInt))
+                .orElseThrow(() ->
+                        new RuntimeException("Can't get min value from list:" + numbers)));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(ZERO_NUMBER, numbers.size())
+                .map(i -> i % CHECKING_NUMBER != ZERO_NUMBER
+                        ? numbers.get(i) - SUBTRACT_NUMBER : numbers.get(i))
+                .filter(n -> n % CHECKING_NUMBER != ZERO_NUMBER)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(f -> f.getSex().equals(Person.Sex.MAN)
+                        && f.getAge() >= fromAge
+                        && f.getAge() <= toAge)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex().equals(Person.Sex.MAN)
```

Comment:
```suggestion
                .filter(p -> p.getSex() == Person.Sex.MAN
```
it's better to compare enums with == not equals
replace it everywhere